packages/gatsby/src/redux/nodes.ts

```diff
@@ -15,12 +15,18 @@ export interface IFilterCache {
   // In this set, `undefined` values represent nodes that did not have the path
```

Comment:
This looks like it probably should be smth like:

```
type FilterCache = {
  op: "eq",
  byValue: Map<FilterValueNullable, Set<IGatsbyNode>>
} | {
 op: "lt",
 valueRangesAsc: Array<IGatsbyNode>
}```

and so on.

Reply:
I kind of like this but I'm not sure how to enforce it without becoming a burden. Especially since the asc/desc arrays will be shared between lte/lt/ne/nin and between gte/gt and won't exist for eq. I'll try to give this some more thought, also with Blaine's feedback.

Reply:
So something like this is definitely possible. But propagating that through the system will require a lot more and clunkier code than it is now. I'm not sure it's very helpful so I propose that we don't do this. At least now now. We can revisit it and have a discussion around the merits for it.

```ts
export type FilterCache =
  | {
      op: "$eq"
      // In this set, `undefined` values represent nodes that did not have the path
      byValue: Map<FilterValueNullable, Set<IGatsbyNode>>
      meta: {}
    }
  | {
      op: "$lte"
      // In this set, `undefined` values represent nodes that did not have the path
      byValue: Map<FilterValueNullable, Set<IGatsbyNode>>
      meta: {
        // Ordered set of all values (by `<`) found by this filter. No null / undefs
        valuesAsc: Array<FilterValue> | void
        // Flat set of nodes, ordered by valueAsc, but not ordered per value group
        nodesByValueAsc: Array<IGatsbyNode> | void
        // Ranges of nodes per value, maps to the nodesByValueAsc array
        valueRangesAsc: Map<FilterValue, [number, number]> | void
      }
    }
  | {
      op: "$gte"
      // In this set, `undefined` values represent nodes that did not have the path
      byValue: Map<FilterValueNullable, Set<IGatsbyNode>>
      meta: {
        // Ordered set of all values (by `>`) found by this filter. No null / undefs
        valuesDesc: Array<FilterValue> | void
        // Flat set of nodes, ordered by valueDesc, but not ordered per value group
        nodesByValueDesc: Array<IGatsbyNode> | void
        // Ranges of nodes per value, maps to the nodesByValueDesc array
        valueRangesDesc: Map<FilterValue, [number, number]> | void
      }
    }
```

Reply:
OK, let's revisit it next time we get a new cache type.